src/erlcloud_aws.erl

```diff
@@ -161,8 +165,8 @@ aws_request4_no_update(Method, Protocol, Host, Port, Path, Params, Service, #aws
                     [{"host", Host}], list_to_binary(Query),
                     Region, Service, [])
     end,
-
-    aws_request_form(Method, Protocol, Host, Port, Path, Query, SignedHeaders, Config).
+    aws_request_form(Method, Protocol, Host, Port, Path, Query,
+                     [SignedHeaders | Headers], Config).
```

Comment:
Why not just properly `++` here, and dispense the unnecessary `lists:flatten/1` below?